schematic/store/synapse.py

```diff
@@ -490,8 +584,8 @@ def getStorageProjects(self, project_scope: List = None) -> list[tuple[str, str]
         # prepare a return list of project IDs and names
         projects = []
         for projectId in storageProjects:
-            projectName = self.syn.get(projectId, downloadFile=False).name
-            projects.append((projectId, projectName))
+            project_name_from_project_header = project_id_to_name_dict.get(projectId)
+            projects.append((projectId, project_name_from_project_header))
```

Comment:
There are a few things to note here on these changes:

1. I have added a change to SYNPY (https://github.com/Sage-Bionetworks/synapsePythonClient/pull/1130) that stored the ID of the user profile that logged into Synapse. This removes the need to call to `getUserProfile()` and saves an extra HTTP call.
2. I am storing the result of getting the paginated list of project headers into the `synapse_entity_tracker`. It is really slow to crawl this information and it occurs several times. By doing this the content is re-used in the context of the request.
3. I removed the need to call `syn.get()` for each of the projects as the paginated list of project headers contained all the data we needed.